src/maintagent/agent/process.c

```diff
@@ -351,3 +352,95 @@ FUNCTION void reIndexAllTables()
   fo_scheduler_heart(1);  // Tell the scheduler that we are alive and update item count
   return;  // success
 }
+
+/**
+ * @brief remove orphaned rows from fossology database
+ * @returns void but writes status to stdout
+ */
+FUNCTION void removeOrphanedRows()
+{
+  PGresult* result; // the result of the database access
+  char *uploadTreeOrphaned;
+  char *clearingDecisionOrphaned;
+  char *clearingDecisionEventOrphaned;
+  char *clearingEventOrphaned;
+  char *obligationMapOrphaned;
+  char *obligationCandidateMapOrphaned;
+  long startTime, endTime;
+
+  char *sql1 = " DELETE FROM uploadtree "
+                 " WHERE uploadtree_pk IN (SELECT DISTINCT(uploadtree_pk) "
+               " FROM uploadtree UT LEFT OUTER JOIN upload U ON U.upload_pk = UT.upload_fk "
+                 " WHERE U.upload_pk IS NULL); ";
```

Comment:
This query can be further optimized like this:

```sql
DELETE FROM uploadtree AS UT
WHERE NOT EXISTS (
  SELECT 1
  FROM upload AS U
  WHERE UT.upload_fk = U.upload_pk
);
```

And same goes for every query below.

I got a Execution time boost (calculated using `EXPLAIN ANALYZE`) from 211,327ms to 32,318ms.